File: cloudtimes-ct/src/main/java/com/cloudtimes/resources/service/impl/CtRegionServiceImpl.java
package com.cloudtimes.resources.service.impl;

import java.util.List;
import com.cloudtimes.common.utils.DateUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.cloudtimes.common.annotation.DataSource;
import com.cloudtimes.common.enums.DataSourceType;
import com.cloudtimes.resources.mapper.CtRegionMapper;
import com.cloudtimes.resources.domain.CtRegion;
import com.cloudtimes.resources.service.ICtRegionService;

/**
 * 地区信息Service业务层处理
 * 
 * @author tank
 * @date 2023-01-17
 */
@DataSource(DataSourceType.CT)
@Service
public class CtRegionServiceImpl implements ICtRegionService 
{
    @Autowired
    private CtRegionMapper ctRegionMapper;

    /**
     * 查询地区信息
     * 
     * @param id 地区信息主键
     * @return 地区信息
     */
    @Override
    public CtRegion selectCtRegionById(Long id)
    {
        return ctRegionMapper.selectCtRegionById(id);
    }

    /**
     * 查询地区信息列表
     * 
     * @param ctRegion 地区信息
     * @return 地区信息
     */
    @Override
    public List<CtRegion> selectCtRegionList(CtRegion ctRegion)
    {
        return ctRegionMapper.selectCtRegionList(ctRegion);
    }

    /**
     * 新增地区信息
     * 
     * @param ctRegion 地区信息
     * @return 结果
     */
    @Override
    public int insertCtRegion(CtRegion ctRegion)
    {
        ctRegion.setCreateTime(DateUtils.getNowDate());
        return ctRegionMapper.insertCtRegion(ctRegion);
    }

    /**
     * 修改地区信息
     * 
     * @param ctRegion 地区信息
     * @return 结果
     */
    @Override
    public int updateCtRegion(CtRegion ctRegion)
    {
        ctRegion.setUpdateTime(DateUtils.getNowDate());
        return ctRegionMapper.updateCtRegion(ctRegion);
    }

    /**
     * 批量删除地区信息
     * 
     * @param ids 需要删除的地区信息主键
     * @return 结果
     */
    @Override
    public int deleteCtRegionByIds(Long[] ids)
    {
        return ctRegionMapper.deleteCtRegionByIds(ids);
    }

    /**
     * 删除地区信息信息
     * 
     * @param id 地区信息主键
     * @return 结果
     */
    @Override
    public int deleteCtRegionById(Long id)
    {
        return ctRegionMapper.deleteCtRegionById(id);
    }
}

File: cloudtimes-ct/src/main/java/com/cloudtimes/resources/service/impl/CtRegionServiceImpl.kt
package com.cloudtimes.resources.service.impl

import com.cloudtimes.common.annotation.DataSource
import com.cloudtimes.common.enums.DataSourceType
import com.cloudtimes.resources.domain.CtRegion
import com.cloudtimes.resources.dto.response.CtRegionResponse
import com.cloudtimes.resources.mapper.CtRegionMapper
import com.cloudtimes.resources.mapper.provider.CtRegionProvider
import com.cloudtimes.resources.service.ICtRegionService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

/**
 * 地区信息Service业务层处理
 *
 * @author tank
 * @date 2023-01-17
 */
@DataSource(DataSourceType.CT)
@Service
class CtRegionServiceImpl : ICtRegionService {
    @Autowired
    private lateinit var ctRegionMapper: CtRegionMapper

    /** 查询地区树 */
    override fun selectCtRegionTree(): List<CtRegionResponse> {
        fun helper(currentList: List<CtRegionResponse>, allList: List<CtRegionResponse>) {
            if (currentList.isEmpty()) {
                return
            }

            for (item in currentList) {
                val children = allList.filter { it.parentRegionCode == item.regionCode }
                if (children.isNotEmpty()) {
                    item.children = children
                    helper(children, allList)
                }
            }
        }

        val regionList = ctRegionMapper.selectMany(CtRegionProvider.selectAllRegions())
            .map { it.toRegionResponse() }
        val level1Regions = regionList.filter { it.regionLevel == "1" }
        helper(level1Regions, regionList)
        return level1Regions
    }

    /**
     * 查询地区信息
     *
     * @param regionCode 地区信息主键
     * @return 地区信息
     */
    override fun selectCtRegionByCode(regionCode: String): CtRegion? {
        return ctRegionMapper.selectCtRegionByCode(regionCode)
    }

    /**
     * 查询地区信息列表
     *
     * @param ctRegion 地区信息
     * @return 地区信息
     */
    override fun selectCtRegionList(ctRegion: CtRegion): List<CtRegion> {
        return ctRegionMapper.selectCtRegionList(ctRegion)
    }

    /**
     * 获取地区名
     *
     * @param regionCode 地区号
     * @return 地区名
     */
    override fun getRegionName(regionCode: String): String? {
        val selectCtRegionByCode = ctRegionMapper.selectCtRegionByCode(regionCode)
        if (selectCtRegionByCode != null) {
            return selectCtRegionByCode.regionName
        }
        return null
    }

}